client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -39,20 +38,28 @@ const resultsSpeak = debounceSpeak( { message: 'Search results loaded.' } );
 const errorSpeak = debounceSpeak( { message: 'No search results found.' } );
 
 function HelpSearchResults( {
-	currentUserId,
 	externalLinks = false,
-	hasAPIResults = false,
-	hasPurchases,
-	isSearching = false,
 	onSelect,
 	onAdminSectionSelect = noop,
 	searchQuery = '',
-	searchResults = [],
-	sectionName,
-	translate,
 	placeholderLines,
-	track,
 } ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const currentUserId = useSelector( ( state ) => getCurrentUserId( state ) );
+	const hasPurchases = useSelector( ( state ) =>
+		hasCancelableUserPurchases( state, getCurrentUserId( state ) )
+	);
+	const sectionName = useSelector( ( state ) => getSectionName( state ) );
+	const contextualResults = useSelector( ( state ) => getContextualHelpResults( state ) );
+	const adminResults = useSelector( ( state ) => getAdminHelpResults( state, searchQuery, 3 ) );
+
+	const { data: searchResults = [], isLoading: isSearching = false } = useInlineHelpSearchQuery(
```

Comment:
The `isLoading` field should always be a valid boolean and doesn't need a default.